pkg/k8sutil/k8sutil.go

```diff
@@ -31,19 +32,23 @@ func WaitForTPRReady(restClient *rest.RESTClient, tprGroup, tprVersion, tprName
 		res := restClient.Get().AbsPath("apis", tprGroup, tprVersion, tprName).Do()
 		err := res.Error()
 		if err != nil {
+			// RESTClient returns *errors.StatusError for any status codes < 200 or > 206
+			// and http.Client.Do errors are returned directly.
+			if se, ok := err.(*errors.StatusError); ok {
+				if se.Status().Code == http.StatusNotFound {
+					return false, nil
+				}
+			}
 			return false, err
 		}
 
 		var statusCode int
 		res.StatusCode(&statusCode)
-		switch statusCode {
-		case http.StatusOK:
-			return true, nil
-		case http.StatusNotFound: // not set up yet. wait.
```

Comment:
Is this not the Status Code that is returned when it is not setup yet? Because waiting is cancelled on error as far as I understand.